elastic/detection-rules
# Add Unit Test test_index_or_data_view_id_present

# Pull Request

*Issue link(s)*: https://github.com/elastic/detection-rules/issues/3965

## Summary - What I changed

- Of the two possible solutions proposed in https://github.com/elastic/detection-rules/issues/3965, the PR is raised for unit test creation.
- Primarily because skipping Unit tests in DAC is simpler and documented [here](https://dac-reference.readthedocs.io/en/latest/internals_of_the_detection_rules_repo.html#unit-testing)
    - With this for developers working on prebuilt rule set, having an index or a data_view_id is enforced.
    - For customers working on their custom rule set, this enforcement can easily be skipped.

## How To Test

- All Unit Tests should Pass
- Local Testing 

> - All rules have the required index
> ```
> Received JSON data in run script
> Running pytest with args: ['-p', 'vscode_pytest', '--rootdir=/Users/shashankks/elastic_workspace/detection-rules', '/Users/shashankks/elastic_workspace/detection-rules/tests/test_all_rules.py::TestValidRules::test_index_or_data_view_id_present']
> ============================= test session starts ==============================
> platform darwin -- Python 3.12.3, pytest-8.1.1, pluggy-1.4.0
> rootdir: /Users/shashankks/elastic_workspace/detection-rules
> configfile: pyproject.toml
> plugins: typeguard-3.0.2
> collected 1 item
> 
> tests/test_all_rules.py .                                                [100%]
> 
> ============================== 1 passed in 49.43s ==============================
> Finished running tests!
> ```

> - Identifying missing indexes like an example case [here](https://github.com/elastic/detection-rules/pull/3953/files) 
> ```
> Received JSON data in run script
> Running pytest with args: ['-p', 'vscode_pytest', '--rootdir=/Users/shashankks/elastic_workspace/detection-rules', '/Users/shashankks/elastic_workspace/detection-rules/tests/test_all_rules.py::TestValidRules::test_index_or_data_view_id_present']
> ============================= test session starts ==============================
> platform darwin -- Python 3.12.3, pytest-8.1.1, pluggy-1.4.0
> rootdir: /Users/shashankks/elastic_workspace/detection-rules
> configfile: pyproject.toml
> plugins: typeguard-3.0.2
> collected 1 item
> 
> tests/test_all_rules.py F                                                [100%]
> 
> =================================== FAILURES ===================================
> ______________ TestValidRules.test_index_or_data_view_id_present _______________
> 
> >           self.fail(fail_msg + '\n'.join(failures))
> E           AssertionError: 
> E                       The following prebuilt rules do not have either 'index' or 'data_view_id' 
> E           
> E                       30b5bb96-c7db-492c-80e9-1eab00db580b - AWS S3 Object Versioning Suspended -> does not have either index or data_view_id
> 
> tests/test_all_rules.py:213: AssertionError
> =========================== short test summary info ============================
> FAILED tests/test_all_rules.py::TestValidRules::test_index_or_data_view_id_present
> ============================== 1 failed in 48.31s ==============================
> Finished running tests!
> ```

## Checklist




## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?
